x-pack/plugins/fleet/server/services/epm/archive/validation.ts

```diff
@@ -15,12 +15,29 @@ import {
   RegistryVarsEntry,
 } from '../../../../common/types';
 import { PackageInvalidArchiveError } from '../../../errors';
-import { pkgToPkgKey } from '../registry';
-import { cacheGet } from './cache';
+import { ArchiveEntry, pkgToPkgKey } from '../registry';
+
+const MANIFESTS: Record<string, Buffer> = {};
+const MANIFEST_NAME = 'manifest.yml';
 
 // TODO: everything below performs verification of manifest.yml files, and hence duplicates functionality already implemented in the
 // package registry. At some point this should probably be replaced (or enhanced) with verification based on
 // https://github.com/elastic/package-spec/
+export async function parseAndVerifyArchiveEntries(
+  entries: ArchiveEntry[]
+): Promise<{ paths: string[]; archivePackageInfo: ArchivePackage }> {
+  const paths: string[] = [];
+  entries.forEach(({ path, buffer }) => {
+    paths.push(path);
+    if (path.endsWith(MANIFEST_NAME) && buffer) MANIFESTS[path] = buffer;
+  });
+
+  return {
+    archivePackageInfo: parseAndVerifyArchive(paths),
+    paths,
+  };
+}
+
 export function parseAndVerifyArchive(paths: string[]): ArchivePackage {
```

Comment:
`parseAndVerifyArchive` needs to read any `manifest.yml` files in the archive to do its validation. Instead of sharing the existing memory cache, use a local map. It only holds a few entries only needs them while parsing. Now it's not coupled to any other implementation or order. It only needs `ArchiveEntry[]`.